ROI startup

ericonr opened the pull request.

ericonr committed:
TLBC2App: add tracing to handle_tlbc2_err.

This is an easy way to add the option of tracing TLBC2 library function
calls, since most of them go through this error handling function.

ericonr committed:
TLBC2App: document TLBC2_get_roi limitations.

ericonr committed:
TLBC2App: fix application of autosave for ROI.

We want the {Min,Size}{X,Y} records to be processed during startup, in
order to apply the ROI stored in autosave. This is especially relevant
because the hardware itself doesn't keep the whole ROI definition, so
restarting the IOC will reset the ROI in the X dimension.

This makes it necessary to use PINI=YES, since these records don't have
it by default, unlike what b5cadf7 (TLBC2App: implement ROI definition
parameters., 2024-07-02) claimed indirectly when it added PINI=NO. The
limitation mentioned in that commit has been overcome here using the
PHAS record field, which enables us to specify that the Size records
must be processed before the Min records, therefore avoiding some
intermediary (invalid) state where the ROI goes out of the image area.

henriquesimoes commented on TLBC2App/Db/ADTLBC2.template:
```diff
@@ -28,19 +28,23 @@ record("*", "$(P)$(R)Gain") {
 }
 
 record("*", "$(P)$(R)MinX") {
-    field(PINI, "NO")
+    field(PINI, "YES")
+    field(PHAS, 2)
 }
 
 record("*", "$(P)$(R)MinX") {
```
It is missing (in another commit?) this typo fix: `s/MinX/MinY/`.

ericonr replied:
Cheers, I missed it.

ericonr replied:
I fixed this locally; will request write permissions to the repo (lost after the transfer) to be able to finish it.

henriquesimoes reviewed (changes requested):
I'd include another fix here, as commented below. Other than that, the set of changes looks good. Thanks!

ericonr committed:
TLBC2App: fix typo in MinY record name.

The MinY record was already configured with PINI=NO, so this mistake
didn't have any effect. Having an actual effect in this overridden
record will happen in a follow-up commit,

Fixes: b5cadf7 (TLBC2App: implement ROI definition parameters., 2024-07-02)

henriquesimoes reviewed (approved):
That looks fine now. Thanks!
